### PR TITLE
Remove `verify_accounts` configuration option

### LINKED ISSUE #2297: Remove verify_accounts config option and VerifyDeployCryptographicValidity event from deploy acceptor
@darthsiroftardis pls add description

### PR DESCRIPTION
**Pull request by Fraser999**
This PR removes the option for operators to disable account verification for deploys received from clients.

It is essentially #2714 rebased on the current tip of `feat-fast-sync`, with outstanding review comments addressed (mainly the removal of the `DeployAcceptor::verify_accounts` member variable) and with the failing tests fixed up.

Closes #2297.

### CODE REVIEW

File: node/src/reactor/participating.rs [hunk 1]
```diff
@@ -622,7 +622,7 @@ impl reactor::Reactor for Reactor {
             NodeState::Participating,
         )?;
 
-        let deploy_acceptor = DeployAcceptor::new(config.deploy_acceptor, &*chainspec, registry)?;
+        let deploy_acceptor = DeployAcceptor::new(&*chainspec_loader.chainspec(), registry)?;
```
Comment (rafal-ch) [inline node/src/reactor/participating.rs:625]:
```suggestion
        let deploy_acceptor = DeployAcceptor::new(chainspec_loader.chainspec(), registry)?;
```

File: node/src/reactor/joiner.rs [hunk 2]
```diff
@@ -622,11 +622,7 @@ impl reactor::Reactor for Reactor {
 
         let trie_or_chunk_fetcher = fetcher_builder.build("trie_or_chunk")?;
 
-        let deploy_acceptor = DeployAcceptor::new(
-            config.deploy_acceptor,
-            &*chainspec_loader.chainspec(),
-            registry,
-        )?;
+        let deploy_acceptor = DeployAcceptor::new(&*chainspec_loader.chainspec(), registry)?;
```
Comment (rafal-ch) [inline node/src/reactor/joiner.rs:625]:
```suggestion
        let deploy_acceptor = DeployAcceptor::new(chainspec_loader.chainspec(), registry)?;
```

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Comment by Fraser999** [comment]
bors merge

**Review by fizyk20 — APPROVED** [review]
LGTM :+1:

**Pull request merged**